code/utils/io/FileWriter.py

```diff
@@ -0,0 +1,67 @@
+import logging as LOG
+from typing import Dict
+
+import pandas as pd
+from common.Constants import META_HEADERS, METRICS_HEADERS
+from customexceptions.ReadabilityErrors import IOError
+from models.CoreMetrics import CoreMetrics
+from models.OutputDataModel import OutputDataModel
+from models.ReadabilityMetrics import ReadabilityMetrics
+
+
+class FileWriter:
+    def __init__(self):
+        """
+        File Writer init
+        """
+        LOG.debug(f'{__name__} init')
+        self.outputCoreMetricsDf = None
+        self.outputReadabilityMetricsDf = None
+
+    def __generateCoreMetricsDf(self, coreMetrics: CoreMetrics) -> pd.DataFrame :
```

Comment:
would suggest to have all the generate metrics method in a separate class

Reply:
I agree. Just want to check on where to place it.
Can we merge this and then refactor? 
Will be adding tests as well before closure of the issue.

Reply:
yes, definitely